features/call/impl/src/main/kotlin/io/element/android/features/call/impl/ui/CallScreenView.kt

```diff
@@ -151,14 +157,28 @@ private fun CallWebView(
             Text("WebView - can't be previewed")
         }
     } else {
+        var audioDeviceCallback: AudioDeviceCallback? by remember { mutableStateOf(null) }
         AndroidView(
             modifier = modifier,
             factory = { context ->
                 // Set 'voice call' mode so volume keys actually control the call volume
-                val audioManager = context.getSystemService<AudioManager>()
-                audioManager?.mode = AudioManager.MODE_IN_COMMUNICATION
+                context.getSystemService<AudioManager>()?.let { audioManager ->
+                    audioManager.mode = AudioManager.MODE_IN_COMMUNICATION
+                    audioManager.enableExternalAudioDevice()
+                    audioDeviceCallback = object : AudioDeviceCallback() {
+                        override fun onAudioDevicesAdded(addedDevices: Array<out AudioDeviceInfo>?) {
+                            Timber.d("Audio devices added")
+                            audioManager.enableExternalAudioDevice()
+                        }
 
-                audioManager?.enableExternalAudioDevice()
+                        override fun onAudioDevicesRemoved(removedDevices: Array<out AudioDeviceInfo>?) {
+                            Timber.d("Audio devices removed")
+                            audioManager.enableExternalAudioDevice()
+                        }
+                    }
+
+                    audioManager.registerAudioDeviceCallback(audioDeviceCallback, null)
+                }
```

Comment:
Could we move this to an external function? Also, maybe we should check if the selected audio device was actually set after selecting it and retry if it's not. I'm not sure if there's some error/success callback for `setCommunicationDevice`.

Reply:
`setCommunicationDevice` returns a boolean and can also throw Exception, so yes, maybe there is some improvement to do here. Let me check.